sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1635,6 +1649,46 @@ static bool setEnvVar(const char *name, const char *value) {
   return true;
 }
 
+static class ZeUSMImport {
+  // Pointers to functions that import/release host memory into USM
+  ze_result_t (*zexDriverImportExternalPointer)(ze_driver_handle_t hDriver,
+                                                void *, size_t);
+  ze_result_t (*zexDriverReleaseImportedPointer)(ze_driver_handle_t, void *);
+  bool ImportSupported;
+
+public:
+  ZeUSMImport() : ImportSupported{false} {}
+  void setZeUSMImport(pi_platform Platform) {
+    // Check if USM hostptr import feature is available.
+    ze_driver_handle_t driverHandle = Platform->ZeDriver;
+    bool ImportSupported =
+        ZE_CALL_NOCHECK(
+            zeDriverGetExtensionFunctionAddress,
+            (driverHandle, "zexDriverImportExternalPointer",
+             reinterpret_cast<void **>(&zexDriverImportExternalPointer))) == 0;
+    if (ImportSupported) {
+      ZE_CALL_NOCHECK(
+          zeDriverGetExtensionFunctionAddress,
+          (driverHandle, "zexDriverReleaseImportedPointer",
+           reinterpret_cast<void **>(&zexDriverReleaseImportedPointer)));
+      // Turn on hostptr import only if supported and explicitly requested.
+      USMHostPtrImportEnabled = true;
+      // Hostptr import is only possible if piMemBufferCreate receives a
+      // hostptr as an argument. The SYCL runtime does that only when
+      // SYCL_HOST_UNIFIED_MEMORY is enabled. Therefore we turn it on.
+      setEnvVar("SYCL_HOST_UNIFIED_MEMORY", "1");
+    }
+  }
+  void doZeUSMImport(ze_driver_handle_t driverHandle, void *HostPtr,
+                     size_t Size) {
+    ZE_CALL_NOCHECK(zexDriverImportExternalPointer,
+                    (driverHandle, HostPtr, Size));
+  }
+  void doZeUSMRelease(ze_driver_handle_t driverHandle, void *HostPtr) {
+    ZE_CALL_NOCHECK(zexDriverReleaseImportedPointer, (driverHandle, HostPtr));
+  }
+} ZeUSMImport;
```

Comment:
Three statics `ZeUSMImport`, `USMHostPtrImportEnabled`, and `USMHostPtrImport` are added, where only one is really needed. Please refactor into a single one (extension is supported and going to be used).

Reply:
The zeUSMImport struct keeps track of the extension API pointers for Import and Release. These could be moved into the _pi_platform struct getting rid of this static struct. Please comment if that is preferred.

Reply:
I actually think it's good that specific dealing with the extension is in its separate class.

Reply:
So I'll keep the zeUSMImport class as is, but renamed zeUSMImportExtension, as you requested earlier.
And you would like to combine USMHostPtrImportEnabled, and USMHostPtrImport?
One of them is whether the env var is set. The other is the final setting, whether the feature is going to be used. It seems like it is best to keep all three.

Reply:
I suggest moving functionalities of USMHostPtrImportEnabled, and USMHostPtrImport into zeUSMImportExtension